### PR TITLE
test(react-draggable-dialog): remove forced mocks

### PR DESCRIPTION
**Pull request by marcosmoura**
A quick fix removing forced mocks for the entire react-components package

### CODE REVIEW

File: packages/react-draggable-dialog/src/components/DraggableDialogHandle/DraggableDialogHandle.test.tsx [hunk 1]
```diff
@@ -7,19 +7,15 @@ import * as contexts from '../../contexts/DraggableDialogContext';
...
-
-jest.mock('@dnd-kit/core', () => {
-  return {
-    __esModule: true,
-    ...jest.requireActual('@dnd-kit/core'),
-  };
-});
+jest.mock('../../contexts/DraggableDialogContext', () => ({
+  __esModule: true,
+  ...jest.requireActual('../../contexts/DraggableDialogContext'),
```
Comment (Hotell) [inline packages/react-draggable-dialog/src/components/DraggableDialogHandle/DraggableDialogHandle.test.tsx:12]: why is this needed ? mocking whole module and using real implementation ?

File: packages/react-draggable-dialog/src/components/DraggableDialog/DraggableDialog.test.tsx [hunk 2]
```diff
@@ -5,18 +5,9 @@ import { DraggableDialog } from './DraggableDialog';
 import { useDraggableDialog } from './useDraggableDialog';
 import { DragEndEvent } from '@dnd-kit/core';
 
-jest.mock('@fluentui/react-components', () => ({
```
Comment (Hotell) [inline packages/react-draggable-dialog/src/components/DraggableDialog/DraggableDialog.test.tsx:8]: noice

**Review by ValentinaKozlova — APPROVED** [review]

File: packages/react-draggable-dialog/src/components/DraggableDialogHandle/DraggableDialogHandle.test.tsx (hunk 1 above)
Comment (marcosmoura) [reply packages/react-draggable-dialog/src/components/DraggableDialogHandle/DraggableDialogHandle.test.tsx:12]: This is needed to be able to Spy on functions that come from that module. Whatever Jest does, hoisting the variables when compiling the test files, doesn't work with ES Modules.

Comment (Hotell) [reply packages/react-draggable-dialog/src/components/DraggableDialogHandle/DraggableDialogHandle.test.tsx:12]: if we really sure we need such amount of mocks etc, we should be explicit.

Take a look at this https://github.com/microsoft/fluentui-contrib/pull/129/commits/c68318d055eee5b00fa2ac9c46ffff99668d3d68

we can talk about it more offline

Comment (marcosmoura) [reply packages/react-draggable-dialog/src/components/DraggableDialogHandle/DraggableDialogHandle.test.tsx:12]: This was added before the availability of the E2E tests. I'm cancelling this PR and I'll those tests in a separate PR in favor of the E2E tests.
